### PR TITLE
Feature/reuse genesis protoc and others

### PR DESCRIPTION
**Pull request by Velenir**
- [x] Reuse GenesisProtocol (will break if on networksother than dev and kovan)
- [x] Grab WETH address from util-contracts
- [x] Grab GEN contract from somewhere (I'm considering `GenesisProtocol.stakingToken()`)

### CODE REVIEW

File: migrations/7_VOTING_MACH-Genesis_Protocol.js [hunk 1]
```diff
@@ -2,37 +2,42 @@
...
+
+
+module.exports = async function (deployer, network) {
+  const GenesisProtocol = artifacts.require('GenesisProtocol')
   const GenToken = artifacts.require('GenToken')
 
   // Deploy Genesis Protocol voting machine
-  // CAN ALSO REUSE
-  await deployGenesisProtocol(DxGenesisProtocol, GenToken, deployer)
+  if (network === 'development') {
```
Comment (anxolin) [inline migrations/7_VOTING_MACH-Genesis_Protocol.js]: I think is better if we move this local development logic to the migration 2 with the other `development` specific contracts, and here we can just assume that `.deployed()` will work (in dev was deployed in 2, in other networks will use the address.

Does it make sense?

File: src/helpers/getDXContractAddresses.js [hunk 2]
```diff
@@ -17,7 +18,7 @@ module.exports = (web3, artifacts) => async (ContractName, dev = USE_DEV_CONTRAC
     ? '@gnosis.pm/dx-contracts/networks-dev.json'
     : '@gnosis.pm/dx-contracts/networks.json'
 
-  const networksJSON = Object.assign(require(networksFile), gnoNetworksJSON)
+  const networksJSON = Object.assign(require(networksFile), gnoNetworksJSON, wethNetworksJSON)
```
Comment (anxolin) [inline src/helpers/getDXContractAddresses.js]: Now we get WETH from utils automatically. It's already merged in add-migrations branch

File: test/contracts/EthToken.sol [hunk 3]
```diff
@@ -2,7 +2,7 @@ pragma solidity ^0.4.24;
 
 import "./TestToken.sol";
 
-contract WethToken is TestToken {
+contract EtherToken is TestToken {
```
Comment (anxolin) [inline test/contracts/EthToken.sol]: I think we don't need this anymore.
This is pooled already in `import "@gnosis.pm/dx-contracts/contracts/DxDevDependencies.sol";`

File: migrations/2_DEV_local_development.js [hunk 4]
```diff
@@ -21,9 +21,10 @@ module.exports = async function (deployer, network, accounts) {
 }
 
 async function deployTokens(deployer, owner) {
+  // TODO: get address from config/networks
   const GenToken = artifacts.require('GenToken') // GEN (Dao Stack)
   const MgnToken = artifacts.require('TokenFRT') // MGN (Token FRT)
```
Comment (anxolin) [inline migrations/2_DEV_local_development.js]: All the tokens but GEN are deployed in the `migrateDx` that was added in migration 2 of add-migration branch

File: test/contracts/EthToken.sol (hunk 3 above)
Comment (Velenir) [reply test/contracts/EthToken.sol]: We don't but I'd leave token contracts for now. Until wrong MGN migration is fixed.

File: src/helpers/getDXContractAddresses.js (hunk 2 above)
Comment (Velenir) [reply src/helpers/getDXContractAddresses.js]: Let it be as an address getter

**Pull request merged**